dataset/src/main/scala/frameless/implicits.scala

```diff
@@ -32,4 +32,12 @@ object implicits {
     // implicit def floatToDouble[T](col: TypedColumn[T, Float]): TypedColumn[T, Double] = col.cast[Double]
     // implicit def floatToBigDecimal[T](col: TypedColumn[T, Float]): TypedColumn[T, BigDecimal] = col.cast[BigDecimal]
   }
+
+  object injections {
+    implicit val javaBoolean: Injection[java.lang.Boolean, Boolean] = Injection(a => a, b => b)
```

Comment:
Hmm I didn't think about this.  From what I tested, using Optional here doesn't come for free in terms of extra code being generated. My main goal with the Java types here was for completeness and and enrich some of the tests.